PyAssembler/Assembler.py

```diff
@@ -0,0 +1,421 @@
+            return rest_line.pop().lstrip()
+        return asm_line
+
+    def section_handler(self, asm_line, translated_words):
+        """ 
+        Checks if the line consist of a section, and handles the conversion
+        of it's data. Update the (potentially) 2 translated words with the new data.
+        
+        :param asm_line: The assembly code line.
+        :param translated_words: The (potentially) 2 words translated so far.
+        :returns: Whether the line is a section line (and handled) or not.
+        :rtype: boolean
+        """
+        section, *rest_line = asm_line.split(maxsplit=1)
+        
+        if section == Assembler.Sections.STRING.value:
+            translated_words[Assembler.FIRST_WORD] = self.convert_str_to_binary(rest_line.pop())      # remove quotes
+        elif section == Assembler.Sections.DATA.value:
+            argument = rest_line.pop().lstrip()
+            if argument.isnumeric():
+                self.current_address += Assembler.WORD_SIZE
+                translated_words[Assembler.FIRST_WORD] = struct.pack(Assembler.TWO_BYTES_FORMAT, int(argument))
+            else:
+                raise ValueError("The input was not a number at line: " + asm_line)
+        else:
+            return False
+        return True
+
+    def handle_1_param_type(self, translated_words, arg):
+        """ 
+        Handles the 1-type line's parameter and update the cumulative translated words.
+        
+        :param translated_words: The (potentially) 2 words translated so far.
+        :param arg: The parameter of the line.
+        """
+        trans_word1, trans_word2 = translated_words
+        
+        # Find 1st argument and then update relevant bits
+        bin_reg = Assembler.REGISTERS.get(arg)
+        if bin_reg:
+            trans_word1 = trans_word1 | bin_reg << Assembler.REG1_OFFSET
+        else:
+            if arg.isnumeric():
+                trans_word2 = int(arg)
+            else:
+                self.untranslated_exps[self.current_line_idx] = arg
+                trans_word2 = None
+            self.current_address += self.WORD_SIZE
+        
+        translated_words[self.FIRST_WORD] = trans_word1
+        translated_words[self.SECOND_WORD] = trans_word2
+
+    def handle_2_param_type(self, translated_words, arg1, arg2):
+        """ 
+        Handles the 2-type line's parameters and update the cumulative translated words.
+        
+        :param translated_words: The (potentially) 2 words translated so far.
+        :param arg1: The first argument (must be register)
+        :param arg2: The second argument
+        """
+        trans_word1, trans_word2 = translated_words
+        # Find 1st register and then update reg1 bits
+        bin_reg1 = Assembler.REGISTERS.get(arg1)
+        
+        if bin_reg1 is None: 
+            raise Assembler.NotRegisterArgumentException(None, arg1)
+        
+        trans_word1 = trans_word1 | bin_reg1 << Assembler.REG1_OFFSET  

```

Comment:
Parenthesis are recommended here.
(Also, consider creating a function to prevent multiple uses of these bitwise logic)